### ISSUE TITLE
Questions regarding the simulation

### ISSUE THREAD

**Issue by BradKML** [hunar4321/Reaching-global-maximum#1]
1. Is assortative mating and genetic heritability considered (for behaviors)?
2. Can the liberal vs conservative divide be seen as hyperparameter optimization "learning rate"? (see [here](http://www.bdhammel.com/learning-rates/) and [here](https://www.jeremyjordan.me/nn-learning-rate/))
3. Has CSR selection been considered ("r/K selection" as reproductive rate and resource intensity) as an intermediary factor?
4. Are multi-layered cycles considered? It might be that certain behaviors are anti-fragile against the different waves ([K-Wave + Kuznet](https://www.sociostudies.org/almanac/articles/long_wave_rhythms_a_pictorial_guide_to_220_years_of_us_history_with_forecasts/) and [Cliodynamics](https://peterturchin.com/age-of-discord/))

**Comment by hunar4321** [comment]
Hi 
1. Mating is an interesting concept. Neat algorithm has done it, I guess. For the first part of the video I used a simple hill climbing algorithm (no mating). A bare form version of the algorithm in python can be found here: 
https://github.com/hunar4321/genetic-algorithm/blob/master/very_simple_GA.py

I also made a blog explaining the algorithm here: https://www.brainxyz.com/machine-learning/genetic-algorthim/

2. Yes, the steps are similar to the learning rate in SGD
3. No
4. I don't know about that, I'll read about it thanks for the links

**Comment by BradKML** [comment]
Regarding CSR selection, it is based on [Grime & Pierce](https://www.nature.com/articles/s41598-019-42562-7) and later [Winemiller & Rose](https://rethinkpriorities.org/publications/life-history-classification) on how resource scarcity and competition changes selected traits and genetics. I think some [crass "pundit"](https://archive.org/details/BitChute-O7LVOtQdp40) also made similar speculative theories for human behavior. Can the mechanism of fight for resource vs fleeing be created in these simulation as well? https://en.wikipedia.org/wiki/Lotka%E2%80%93Volterra_equations
